include/irr/asset/CTexturePacker.h

```diff
@@ -60,21 +121,13 @@ class CTexturePacker
         auto extent = _img->getCreationParameters().extent;
         uint32_t rnd = core::roundUpToPoT(std::max(extent.width,extent.height))/m_pgSzxy;
         uint32_t pgCnt = rnd*rnd;
-        uint32_t morton_addr = m_addrAlctr.alloc_addr(pgCnt, pgCnt);
-        assert(morton_addr!=addr_alctr_t::invalid_address);
-
-        auto dec_morton = [](uint32_t x) -> uint32_t {
-            x = x & 0x55555555u;
-            x = (x | (x >> 1)) & 0x33333333u;
-            x = (x | (x >> 2)) & 0x0F0F0F0Fu;
-            x = (x | (x >> 4)) & 0x00FF00FFu;
-            x = (x | (x >> 8)) & 0x0000FFFFu;
-            return x;
-        };
-        core::vector2du32_SIMD texOffset(dec_morton(morton_addr), dec_morton(morton_addr >> 1));
-        texOffset *= m_pgSzxy;
+        uint32_t addr = m_addrAlctr.alloc_addr(pgCnt, pgCnt);
+        uint32_t addr_layer = m_addrAlctr.unpackAddress_layer(addr);
+        assert(addr!=addr_alctr_t::invalid_address);
+
```

Comment:
at this point, you want to call a second allocator (the Pool Allocator which has bins of sizeof(uint32_t) and as many bins as you have tiles in the megatexture)

You `multi_alloc` as many tiles as the texture actually has (not the PoT rounded up count), and you need to fill up the page table texels with the Pool Address Allocated addresses.

Reply:
you need to be able to handle allocation failure